File: tools-starter-swagger/src/main/java/cn/gjing/tools/swagger/ResponseSchema.scala
package cn.gjing.tools.swagger

import scala.beans.BeanProperty

/**
 * @author Gjing
 **/
class ResponseSchema {
  /**
   * 状态码
   */
  @BeanProperty
  var code: Int = _
  /**
   * 响应信息
   */
  @BeanProperty
  var message: String = _
  /**
   * 结果Bean名称
   */
  @BeanProperty
  var schema: String = _

  def this(code: Integer, message: String, schema: String) {
    this()
    this.code = code
    this.message = message
    this.schema = schema
  }
}

File: tools-starter-swagger/src/main/java/cn/gjing/tools/swagger/ResponseSchema.java
package cn.gjing.tools.swagger;

import lombok.*;

/**
 * Global response structure
 *
 * @author Gjing
 **/
@Builder
@Setter
@Getter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class ResponseSchema {
    /**
     * Response status
     */
    private String code;
    /**
     * Response message
     */
    private String message;
    /**
     * Result bean name
     */
    @Deprecated
    private String schema;
    /**
     * Whether is default
     */
    @Builder.Default
    private boolean isDefault = false;
}
